cf-monitord/mon_cpu.c

```diff
@@ -62,8 +62,8 @@ void MonCPUGatherData(double *cf_this)
         {
             break;
         }
-
-        if (sscanf(buf, "%s%ld%ld%ld%ld%ld%ld%ld", cpuname, &userticks, &niceticks, &systemticks, &idle, &iowait, &irq,
+        assert(CF_MAXVARSIZE == 1024);
+        if (sscanf(buf, "%1023s%ld%ld%ld%ld%ld%ld%ld", cpuname, &userticks, &niceticks, &systemticks, &idle, &iowait, &irq,
                &softirq) != 8)
```

Comment:
Would be nice if this pattern could be avoided and the CF_MAXVARSIZE and CF_BUFSIZE could be used more directly to avoid typos.

Reply:
I considered a few different approaches (macro, function call, expanding `CF_MAXVARSIZE` inside the format string, etc.). I landed on this one because it is a good compromise between safety, readability and maintainability. The assert forces you to update the format string if you change the constant macro. The format string is still pretty readable, and it does prevent you from overflowing the buffer.

I agree that it is not "super nice", but I think it's the best I could do within the confines of C and `sscanf`.